lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -1284,17 +1301,32 @@ def show_module_set(type, module_set, regex = nil, minrank = nil, opts = nil) #
             print(tbl.to_s)
           end
 
-          def generate_module_table(type, search_term = nil) # :nodoc:
-            Table.new(
-              Table::Style::Default,
-              'Header'     => type,
-              'Prefix'     => "\n",
-              'Postfix'    => "\n",
-              'Columns'    => [ '#', 'Name', 'Disclosure Date', 'Rank', 'Check', 'Description' ],
-              'SearchTerm' => search_term
-            )
+          def generate_module_table(type, search_terms = [], row_filter = nil) # :nodoc:
+              Table.new(
+                Table::Style::Default,
+                'Header'     => type,
+                'Prefix'     => "\n",
+                'Postfix'    => "\n",
+                'Columns'    => [
+                  '#',
+                  'Name',
+                  'Disclosure Date',
+                  'Rank',
+                  'Check',
+                  'Description'
+                ],
+                'SearchTerm' => row_filter,
+                'Formatters' => {
+                  'Rank' => RankFormatter.new,
+                },
+                'Stylers' => {
```

Comment:
To help reduce the possibility of mis-typing the column names between Columns, Formatters, and Stylers, what are your thoughts on changing the API to bring all that information under column definitions directly?

```ruby
  'Columns' => [
    '#',
    'Name',
    'Disclosure Date',
    'Rank' => {
      formatters: [RankFormatter.new],
      stylers: [RankStyler.new],
    },
    'Check',
    'Description' => {
      styles: [HighlightSubstringStyler.new(search_terms)]
    }
  ]
```

That way the code is more DRY, as we don't have to duplicate the column names, and the next reader of the code can find all of the configuration for a particular column in one place - whilst at the minute they will need to look at 3 different sections to see what a column might be styled as

Reply:
In the snippet I've also changed the styles/formatters to take an array, just to future proof our API. It's not strictly needed though - as a [composite](https://en.wikipedia.org/wiki/Composite_pattern) formatter/styler could be introduced - so that one is your call :+1: